src/linker/Linker.Steps/SealerStep.cs

```diff
@@ -0,0 +1,124 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System.Diagnostics;
+using System.Collections.Generic;
+using Mono.Cecil;
+
+namespace Mono.Linker.Steps
+{
+	public class SealerStep : BaseStep
+	{
+		public SealerStep ()
+		{
+		}
+
+		protected override void ProcessAssembly (AssemblyDefinition assembly)
+		{
+			if (Annotations.GetAction (assembly) != AssemblyAction.Link)
+				return;
+
+			if (!Context.Optimizations.IsEnabled (CodeOptimizations.Sealer, assembly))
+				return;
+
+			foreach (var type in assembly.MainModule.Types)
+				ProcessType (type);
+		}
+
+		bool IsSubclassed (TypeDefinition type)
+		{
+			foreach (var a in Context.GetAssemblies ()) {
+				foreach (var s in a.MainModule.Types) {
+					if (s.BaseType?.IsTypeOf (type.Namespace, type.Name) == true)
+						return true;
+
+					if (s.HasNestedTypes) {
+						foreach (var ns in s.NestedTypes) {
+							if (ns.BaseType.IsTypeOf (type.Namespace, type.Name))
```

Comment:
`BaseType` could be null if this is a nested interface

Reply:
Fixed